Exercise_3/Adapters/FileAdapter.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+using Android.App;
+using Android.Content;
+using Android.OS;
+using Android.Runtime;
+using Android.Views;
+using Android.Widget;
+using Java.IO;
+using Object = Java.Lang.Object;
+
+namespace Exercise_3.Adapters
+{
+    class FileAdapter : BaseAdapter
+    {
+        private List<string> Files { get; }
+        private Activity activity { get; }
+
+        public FileAdapter(List<string> files, Activity activity)
+        {
+            Files = files;
+            this.activity = activity;
+        }
+        public override Object GetItem(int position)
+        {
+            return Files[position];
+        }
+
+        public override long GetItemId(int position)
+        {
+            return 0;
+        }
+
+        public override View GetView(int position, View convertView, ViewGroup parent)
+        {
+            var view = convertView ?? activity.LayoutInflater.Inflate(
```

Comment:
Don's store the activity. You just need to do like this
```C#
var view = convertView?? LayoutInflater.From(parent.Context). Inflate(....)
```

Reply:
All done